lib/Myriad/Registry.pm

```diff
@@ -87,7 +87,7 @@ async method add_service (%args) {
     await $srv->load();
     my $k = refaddr($srv);
     weaken($service_by_name{$service_name} = $srv);
-    weaken($myriad->services->{$k} = $srv);
+    weaken($myriad->services->{scalar(keys $myriad->services->%*)} = $srv);
```

Comment:
@nael-binary This change makes no sense, why is it needed?

Reply:
This is needed to maintain the order of services added. Im not sure if `refaddr` would be incremental.